clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -817,10 +819,10 @@ public void onSuccess(OffsetsForLeaderEpochClient.OffsetForEpochResult offsetsRe
                     // In addition, check whether the returned offset and epoch are valid. If not, then we should reset
                     // its offset if reset policy is configured, or throw out of range exception.
                     offsetsResult.endOffsets().forEach((respTopicPartition, respEndOffset) -> {
-                        SubscriptionState.FetchPosition requestPosition = fetchPositions.get(respTopicPartition);
+                        FetchPosition requestPosition = fetchPositions.get(respTopicPartition);
 
                         if (respEndOffset.hasUndefinedEpochOrOffset()) {
-                            handleOffsetOutOfRange(requestPosition.offset, respTopicPartition,
+                            handleOffsetOutOfRange(requestPosition, respTopicPartition,
                                 "Failed leader offset epoch validation for " + respEndOffset
```

Comment:
Seems like it's more useful to reference `requestPosition` here since the end offset is undefined.